### PR TITLE
Move alica test files

### PR DESCRIPTION
**Pull request by Dale-Koenig**
- Move plans/behaviours outside of "test" directory to a more obvious place
- Clean up and fix some cmake/test configuration related stuff

### CODE REVIEW

File: alica_tests/src/test/test_alica_engine_plan_parser.cpp [hunk 1]
```diff
@@ -575,10 +575,9 @@ class MultiConfigFolderTest : public AlicaEngineTest
     std::vector<std::string> getTestFolderPaths() const override
     {
         std::string path;
-#if defined(PLANS)
-        path = std::string{PLANS} + "/src/test";
+#if defined(THIS_PACKAGE_DIR)
+        path = std::string{THIS_PACKAGE_DIR} + "/etc/";
```
Comment (rogeriofonteles) [inline alica_tests/src/test/test_alica_engine_plan_parser.cpp:579]: really nit: `std::string` is not necessary, and this was something I passed on my PR, we can remove it.

File: alica_tests/src/test/test_assignment.cpp [hunk 2]
```diff
@@ -22,9 +22,9 @@ TEST(Assignment, RobotsInserted)
 {
     // Path to test configs set by CMake
     std::string path;
-#if defined(PLANS)
-    path = PLANS;
-    path += "/src/test";
+#if defined(THIS_PACKAGE_DIR)
+    path = THIS_PACKAGE_DIR;
+    path += "/etc/";
```
Comment (rogeriofonteles) [inline alica_tests/src/test/test_assignment.cpp:27]: 1 line maybe?

**Review by rogeriofonteles — APPROVED** [review]
Seems fine. :+1:

File: alica_tests/src/test/test_alica_engine_plan_parser.cpp (hunk 1 above)
Comment (Dale-Koenig) [reply alica_tests/src/test/test_alica_engine_plan_parser.cpp:579]: Actually it is necessary, two `const char*` cannot be added together

File: alica_tests/test/alica_tests.test [hunk 3]
```diff
@@ -1,14 +0,0 @@
-<launch>
-    <arg name="use_gdb" default="$(optenv GDB false)"/>
```
Comment (veerajsk) [inline alica_tests/test/alica_tests.test]: we don't need this file?

**Review by veerajsk — APPROVED** [review]

File: alica_tests/test/alica_tests.test (hunk 3 above)
Comment (Dale-Koenig) [reply alica_tests/test/alica_tests.test]: As far as I can tell this file was no longer used now that the tests are just done with CMake+Gtest skipping catkin

**Pull request merged**